src/components/CustomModal/index.jsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react';
+import { ComposedModal, ModalHeader, ModalBody, TextInput, Button, ModalFooter } from "carbon-components-react";
+import { postRequest, invItemURL } from "../../utils/api-utils";
+import styles from "./customModal.module.scss";
+
+const CustomModal = (props) => {
+    
+    const {
+        data,
+        showModal,
+        rootClass,
+        modalListClass,
+        subTitle,
+        primaryButton,
+        secondaryButton,
+        tabs,
+        handleSubmit,
+        closeModal } = props;
+    const [ modifiedData, setModifiedData ] = useState(data);
+    const [ postError, setPostError ] = useState('');
+    const [ editMode, setEditMode ] = useState({ "id": -1, "value": 0 });
+
+    /**
+     * description. send the post request with updated dispense as payload
+     * @namespace submitDespense
+     * @memberof  CustomModal
+     * @property {array} payload modified dispense data.
+    */
+
+    const submitDespense = async (payload) => {
+        const res = await postRequest(invItemURL, payload)
+        const responseData = await res.json();
+        if(res.error) {
+            setPostError(res.error.message);
+          } else {
+            handleSubmit(responseData);
+            closeModal(false)
+          }
+    }
+
+    /**
+     * description. handle the editable input field and update the dispense data obj
+     * @namespace handleEditMode
+     * @memberof  CustomModal
+     * @property {object} e current click event.
+     * @property {number} id index of modified dispense data.
+    */
+
+    const handleEditMode = (e, id) => {
+        e.preventDefault();
+        const targetValue = e && e.target && e.target.value;
+        if (targetValue && !/^[1-9]\d*(\.\d+)?$/.test(targetValue)) {
+            return false;
+        } else {
+            handleChange(targetValue, id);
+            setEditMode({ "id": id, "value": targetValue });
+        }  
+    }
+
+    /**
+     * description. check if data is modified or not
+     * @namespace isUpdate
+     * @memberof  CustomModal
+     * @property {object} obj orginal dispense data.
+     * @property {object} modifiedObj modified dispense data.
+    */
+    const isUpdate = (obj, modifiedObj) => {
+        return modifiedObj.find((item, index) => parseInt(item.quantity) !== parseInt(obj[index].quantity))
+    }
+    
+    /**
+     * description. Modified the dispense data with updated quantity
+     * @namespace handleChange
+     * @memberof  CustomModal
+     * @property {number} targetValue modified dispense quantity.
+     * @property {number} index index of edit object.
+    */
+    const handleChange = (targetValue, index) => {
+        modifiedData.dispense_drugs = modifiedData?.dispense_drugs?.map((item, i) => (index === i ? { ...item, ['quantity']: parseInt(targetValue) } : item ));
+        setModifiedData(modifiedData) 
+    }
+
+    useEffect(() => {
+        console.log('load modal')
```

Comment:
Is this useEffect() needed?

Reply:
This screen is used in the dispense screen, which was developed with mock data's when UI devs worked on it. Is it fine if we take all the feedback related to dispense screen as part of https://trello.com/c/kLG4SL2k/28-inventory-dispensation-system-dispense-to-patients?